libc/unistd_getopt.c

```diff
@@ -314,6 +313,101 @@ TEST(unistd_getopt, getopt_unknownopt_noarg)
 }
 
 
+TEST(unistd_getopt, getopt_unknownopt_multiple)
+{
+	/* mocks of main() arguments */
+	char *testargv1[] = { "cmd", "-xxx", "", "" };
+	char *testargv2[] = { "cmd", "-ab", "-c", "value" };
+	int testargc1 = 2, testargc2 = 4;
+
+	testmain(testargc1, testargv1, &ret, "abc:");
+	TEST_ASSERT_EQUAL_INT(UNKNOWNOPT, ret.err);
+	TEST_ASSERT_EQUAL_INT(0, ret.aflag);
+	TEST_ASSERT_EQUAL_INT(0, ret.bflag);
+	TEST_ASSERT_NULL(ret.cvalue);
+	TEST_ASSERT_EQUAL_INT(0, ret.nonopts);
+
+	/* Assert that parsing argument works correctly after passing multiple unknown options
+	   There was issue related to this case */
+
+	/* default presets of results structure */
```

Comment:
either provide function for resetting `struct getopt_t` type or use `memset`.

Please note that this doesn't need to be a single test (if the getopt resetting is not working - some of the tests executed after this one would fail

Reply:
Yeah, I agree on that, but I would rather leave it to show by case name the reason of fail